packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+     * Default value is Photo
+     */
+    startMode?: CameraStartMode;
+
+    /**
+     * @hidden
+     * Optional; indicate if user is allowed to move between front and back camera
+     * Default value is true
+     */
+    cameraSwitcher?: boolean;
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10 set by lens-sdk.
+     */
+    visualMediaCount: number;
+  }
+  /**
+   *  All properties in ImageProps are optional and have default values in the platform
+   *  Additional properties for image in mobile devices
+   * @beta
+   */
+  export interface ImageProperties extends VisualMediaProps {
+    /**
+     * Optional; indicate if inking on the selected Image is allowed or not
+     * Default value is true
+     */
+    ink?: boolean;
+
+    /**
+     * Optional; indicate if putting text stickers on the selected Image is allowed or not
+     * Default value is true
+     */
+    textSticker?: boolean;
+
+    /**
+     * Optional; indicate if image filtering mode is enabled on the selected image
+     * Default value is false
+     */
+    enableFilter?: boolean;
+
+    /**
+     * Optional; Lets the developer specify the image output formats, more than one can be specified.
+     * Default value is Image.
+     */
+    imageOutputFormats?: ImageOutputFormats[];
+  }
+
+  /**
+   * The modes in which camera can be launched
+   * @beta
+   */
+  export enum CameraStartMode {
+    /** Photo mode. */
+    Photo = 1,
+    /** Document mode. */
+    Document = 2,
+    /** Whiteboard mode. */
+    Whiteboard = 3,
+    /** Business card mode. */
+    BusinessCard = 4,
+  }
+  /**
+   * Enum for file formats supported
+   * @beta
+   */
+  export enum FileFormat {
+    /** Base64 encoding */
+    Base64 = 'base64',
+    /** File id */
+    ID = 'id',
+  }
+  /**
+   * Specifies the image source
+   * @beta
+   */
+  export enum Source {
+    /** visual media source is camera. */
+    Camera = 1,
+    /** visual media source is gallery. */
+    Gallery = 2,
+  }
+  /**
+   * Specifies the image output formats.
+   * @beta
+   */
+  export enum ImageOutputFormats {
+    /** Outputs image.  */
+    IMAGE = 1,
+    /** Outputs pdf. */
+    PDF = 2,
+  }
+
+  /**
+   * VisualMediaFile object that can be used to represent image or video
+   *
+   * @beta
+   */
+  export class VisualMediaFile {
+    /**
+     * Content of the file. When format is Base64, this is the base64 content
+     * When format is ID, this is id mapping to the URI
+     * When format is base64 and app needs to use this directly in HTML tags, it should convert this to dataUrl.
+     */
+    public content: string;
+
+    /**
+     * Format of the content
+     */
+    public format: FileFormat;
+
+    /**
+     * Size of the file in KB
+     */
+    public size: number;
+
+    /**
+     * MIME type. This can be used for constructing a dataUrl, if needed.
+     */
+    public mimeType: string;
```

Comment:
Some of these properties look like they don't universally apply (e.g. is `mimeType` specified if the filetype is ID? Is `size` specified for ID types?) Unless all the properties can validly apply to both file types, consider using a technique like I suggested above for `VisualMediaProps` to only allow for properties that are valid for each type to be specified, and not allow invalid combinations.

Reply:
size is the field that needed with both file format. I've updated the data model to only allow for properties that are valid for each type to be specified. Thanks for the great suggesting!